DisguiseUnityRenderStream/Runtime/Util/RegistryWrapper.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Disguise.RenderStream.Utils
+{
+    // https://www.pinvoke.net/default.aspx/advapi32/regopenkeyex.html
+    static class RegistryWrapper
+    {
+        public static UIntPtr HKEY_LOCAL_MACHINE = new UIntPtr(0x80000002u);
+        public static UIntPtr HKEY_CURRENT_USER  = new UIntPtr(0x80000001u);
+
+        public enum ReadRegKeyResult
+        {
+            Success,
+            OpenFailed,
+            QueryValueFailed,
+            TypeNotSupported
+        }
+
+        // Only supports REG_SZ registry value type
+        public static ReadRegKeyResult ReadRegKey(UIntPtr rootKey, string keyPath, string valueName, out string value)
+        {
+            value = null;
+            
+            if (Native.RegOpenKeyEx(rootKey, keyPath, 0, Native.KEY_READ, out var hKey) == 0)
+            {
+                uint size = 1024;
+                StringBuilder keyBuffer = new StringBuilder((int)size);
+
+                if (Native.RegQueryValueEx(hKey, valueName, IntPtr.Zero, out var type, keyBuffer, ref size) == 0)
+                {
+                    Native.RegCloseKey(hKey);
+
+                    if (type == Native.REG_SZ)
+                    {
+                        value = keyBuffer.ToString();
+                        return ReadRegKeyResult.Success;
+                    }
+
+                    return ReadRegKeyResult.TypeNotSupported;
+                }
+                
+                Native.RegCloseKey(hKey);
+                return ReadRegKeyResult.QueryValueFailed;
+            }
+
+            return ReadRegKeyResult.OpenFailed;
+        }
+        
+        static class Native
+        {
+            const string advapidll = "advapi32.dll";
+            
+            public const int KEY_READ = 0x20019;
+
+            public const uint REG_NONE = 0x00000000;
+            public const uint REG_SZ = 0x00000001;
+            
+            [DllImport(advapidll, CharSet = CharSet.Auto, SetLastError = true)]
+            public static extern int RegOpenKeyEx(
+                UIntPtr hKey,
+                string subKey,
+                int ulOptions,
+                int samDesired,
+                out UIntPtr hkResult);
+            
+            [DllImport(advapidll, CharSet = CharSet.Auto, SetLastError = true)]
+            public static extern uint RegQueryValueEx(
```

Comment:
Minor: Why RegOpenKeyEx returns an int and RegQueryValueEx a uint?  In MSDN they are both an LSTATUS.

After checking in windows.h header, LSTATUS is defined as:
```c
typedef _Return_type_success_(return==ERROR_SUCCESS) LONG LSTATUS;
```

So I guess int is a "better value".